src/main.js

```diff
@@ -385,24 +412,63 @@ class WebSearchNavigator {
     const getOpt = (key) => {
       return this.options.sync.get(key);
     };
-    this.register(getOpt('nextKey'), () => {
-      if (!getOpt('autoSelectFirst') && this.isFirstNavigation) {
-        this.resultsManager.focus(0);
-        this.isFirstNavigation = false;
-      } else {
-        this.resultsManager.focusNext(getOpt('wrapNavigation'));
-      }
-      return false;
-    });
-    this.register(getOpt('previousKey'), () => {
-      if (!getOpt('autoSelectFirst') && this.isFirstNavigation) {
-        this.resultsManager.focus(0);
-        this.isFirstNavigation = false;
-      } else {
-        this.resultsManager.focusPrevious(getOpt('wrapNavigation'));
-      }
-      return false;
-    });
+    if (!this.searchEngine.gridNavigation) {
+      this.register(getOpt('nextKey'), () => {
+        if (!getOpt('autoSelectFirst') && this.isFirstNavigation) {
+          this.resultsManager.focus(0);
+          this.isFirstNavigation = false;
+        } else {
+          this.resultsManager.focusNext(getOpt('wrapNavigation'));
+        }
+        return false;
+      });
+      this.register(getOpt('previousKey'), () => {
+        if (!getOpt('autoSelectFirst') && this.isFirstNavigation) {
+          this.resultsManager.focus(0);
+          this.isFirstNavigation = false;
+        } else {
+          this.resultsManager.focusPrevious(getOpt('wrapNavigation'));
+        }
+        return false;
+      });
+    } else {
+      this.register(getOpt('nextKey'), () => { // down
+        if (!getOpt('autoSelectFirst') && this.isFirstNavigation) {
+          this.resultsManager.focus(0);
+          this.isFirstNavigation = false;
+        } else {
+          this.resultsManager.focusDown(getOpt('wrapNavigation'));
+        }
+        return false;
+      });
+      this.register(getOpt('previousKey'), () => { // up
+        if (!getOpt('autoSelectFirst') && this.isFirstNavigation) {
+          this.resultsManager.focus(0);
+          this.isFirstNavigation = false;
+        } else {
+          this.resultsManager.focusUp(getOpt('wrapNavigation'));
+        }
+        return false;
+      });
+      this.register(getOpt('navigatePreviousResultPage'), () => { // left
```

Comment:
The previous/next page bindings are currently defined in the `tabs` function of each search engine in `search_engines.js`, for example the last lines in https://github.com/infokiller/web-search-navigator/blob/5c5a7a9e8622b96eb89a39f253c29a5b4b983941/src/search_engines.js#L471-L491

And then the bindings are performed here:

https://github.com/infokiller/web-search-navigator/blob/5c5a7a9e8622b96eb89a39f253c29a5b4b983941/src/main.js#L306-L331

The way this is implemented now, the bindings are actually set twice: first in `initTabsNavigation`, then in `initResultsNavigation`.
To make it clearer, I suggest making the previous/next page bindings in `initResultsNavigation` as well, and change `search_engines.js` accordingly: move the previous/next elements from `tabs` to new two functions `
previousPageButton`/`nextPageButton`.

Reply:
There is no tabs getter in the Youtube class, mainly because there aren't any on the youtube home page or on the youtube search result (although these should be added for easy navigation across different sections of a youtube channel's home page). Accordingly it makes no sense to add the `previousPageButton`/`nextPageButton` for the youtube class.

I agree it would make readablity easier for the other classes. I am happy to make the change for that but do you think that is in the scope of this PR or should I raise one separately after this?

Reply:
Another PR is great, thanks!